rust/cardano-blockchain-types/src/auxdata/metadatum.rs

```diff
@@ -0,0 +1,122 @@
+//! Raw metadata
+
+use std::sync::Arc;
+
+use dashmap::DashMap;
+use minicbor::Decode;
+
+use super::{metadatum_label::MetadatumLabel, metadatum_value::MetadatumValue};
+use crate::conversion::from_saturating;
+
+/// Transaction Metadata
+/// See: <https://github.com/IntersectMBO/cardano-ledger/blob/78b32d585fd4a0340fb2b184959fb0d46f32c8d2/eras/conway/impl/cddl-files/conway.cddl#L519>
+#[derive(Clone, Debug)]
+pub struct Metadata(Arc<MetadataInner>);
+
+#[derive(Clone, Debug)]
+/// Transaction Metadata - Inner
+/// See: <https://github.com/IntersectMBO/cardano-ledger/blob/78b32d585fd4a0340fb2b184959fb0d46f32c8d2/eras/conway/impl/cddl-files/conway.cddl#L519>
+pub struct MetadataInner {
+    /// Sequence the metadatum labels appear in the metadata.
+    seq: Vec<MetadatumLabel>,
+    /// K/V of metadata items.
+    map: dashmap::ReadOnlyView<MetadatumLabel, MetadatumValue>,
```

Comment:
I'm curious: `dashmap` is a concurrent map, right? But as far as I can see, we are only reading values after `Metadata` creation. Am I missing something or is an expansion of API planned?

Reply:
It is concurrent, and behind an Arc, its highly possible for the contents of the Arc to be accessed by multiple threads, hence the necessity for concurrent safe hashmaps.  The `ReadOnlyView` is a read only version of the DashMap, so it is immutable, but still concurrent safe.